kubetest/main.go

```diff
@@ -938,14 +938,21 @@ func prepare(o *options) error {
 	}
 
 	switch o.provider {
-	case "gce", "gke", "kubernetes-anywhere", "node":
+	case "gce", "gke", "node":
 		if err := prepareGcp(o); err != nil {
 			return err
 		}
 	case "aws":
 		if err := prepareAws(o); err != nil {
 			return err
 		}
+	// kubernetes-anywhere jobs are treated as --provider=local
+	case "local":
```

Comment:
i agree. ideas are welcome, of course.

Reply:
I feel this is odd that provider local will need to set up gcp :-/ why provider is not kubernetes-anywhere here?

Reply:
it only does if the deployer is set though.

It needs to because the tests do not support arbitrary providers :grimacing:

Reply:
alternatively the test code should select a no-op provider when it does not know the provider flag value? this would be closer to previous behavior..

Reply:
so references in k/k, like https://github.com/kubernetes/kubernetes/blob/master/cluster/kube-util.sh#L25-L31 also need to be cleaned up?

maybe we should make prepareGcp explicit :-\

Reply:
> I feel this is odd that provider local will need to set up gcp :-/ why provider is not kubernetes-anywhere here?

there is no more "kubernetes-anywhere" provider in the e2e FW anymore.

> alternatively the test code should select a no-op provider when it does not know the provider flag value? this would be closer to previous behavior..

this would require changes in both the e2e and kubetest.

also, what we can do in kubetest is simply call prepareGcp() based on the deployment:
i.e. remove this:
https://github.com/kubernetes/test-infra/blob/c0bfd9bfb245a2902e110dac16a666eaaeb9ccdd/kubetest/main.go#L949-L955
and call prepareGcp() here:
https://github.com/kubernetes/test-infra/blob/c0bfd9bfb245a2902e110dac16a666eaaeb9ccdd/kubetest/main.go#L253

?

Reply:
Er what I mean is, where the e2e framework selects providers, it could default to the no-op provider for any provider for which there is not an actual known entry.

Reply:
@BenTheElder The problem with that approach is that genuine typos (like `--provider=gec` instead of `--provider=gce`) then do not get detected.

Right now the `e2e.test` binaries has two special providers:
- `local` for "run against arbitrary cluster" and do some extra logging
- `skeleton` for "run against arbitrary cluster" without that extra logging

Can't you use `skeleton`? @timothysc was against adding `kubernetes-anywhere` because its the same as `skeleton`.

Reply:
> The problem with that approach is that genuine typos (like --provider=gec instead of --provider=gce) then do not get detected.

before they created a warning in the logs.

> Can't you use skeleton?

Well yes, the problem is other things are aware of which provider you're using... eg the log dump script in k/k's cluster/ dir. Here it's using """local""" but needs to also act like GCP because it's doing kubernetes-anywhere on GCE but not necessarily using the GCE provider.

Reply:
^ @krzyzacy what do you think about calling  prepareGcp() in getDeployer() for the k-a case?

Reply:
let me just update this PR.
(might hide the above conversation).